### PR TITLE
Added possibility to define cookie path parameter #2

### PR DESCRIPTION
**Pull request by IgorKvasn**
replacement for a bad PR #1196

### CODE REVIEW

File: addon/session-stores/cookie.js [hunk 1]
```diff
@@ -148,6 +161,8 @@ export default BaseStore.extend({
   init() {
     this._super(...arguments);
 
+    this.set('_cookiePath', Configuration.baseURL);
```
Comment (marcoow) [inline addon/session-stores/cookie.js]: This should only be set when it is `null` (as it might have been set when the cookie store instance was created with `Cookie.create({ cookiePath: '/whatever' })` and shouldn't be overridden in this case).

File: tests/unit/session-stores/adaptive-test.js [hunk 2]
```diff
@@ -6,13 +6,18 @@ import Adaptive from 'ember-simple-auth/session-stores/adaptive';
...
+import Configuration from 'ember-simple-auth/configuration';
 
 const { assign: emberAssign, merge, run } = Ember;
 const assign = emberAssign || merge;
 
 describe('AdaptiveStore', () => {
   let store;
 
+  beforeEach(function() {
+    Configuration.load({ baseURL: '/' });
```
Comment (marcoow) [inline tests/unit/session-stores/adaptive-test.js]: Why is this required?

File: tests/unit/session-stores/cookie-test.js [hunk 3]
```diff
@@ -17,6 +18,7 @@ describe('CookieStore', () => {
 
   beforeEach(function() {
     store = createCookieStore(FakeCookieService.create());
+    Configuration.load({ baseURL: '/' });
```
Comment (marcoow) [inline tests/unit/session-stores/cookie-test.js]: Why is this required?

File: addon/session-stores/cookie.js (hunk 1 above)
Comment (IgorKvasn) [inline addon/session-stores/cookie.js]: tests in `cookie-store-behaviour.js` expect that cookie path is set to '/' e.g.
https://github.com/simplabs/ember-simple-auth/blob/master/tests/unit/session-stores/shared/cookie-store-behavior.js#L110

I guess there are 2 ways to sort this out - either change it in all asserts from `path === '/'` to `path === Configuration.baseURL` or just set a default baseURL for all tests (which I did)

File: tests/unit/session-stores/adaptive-test.js (hunk 2 above)
Comment (IgorKvasn) [inline tests/unit/session-stores/adaptive-test.js]: tests in `cookie-store-behaviour.js` expect that cookie path is set to '/' e.g.
https://github.com/simplabs/ember-simple-auth/blob/master/tests/unit/session-stores/shared/cookie-store-behavior.js#L110

I guess there are 2 ways to sort this out - either change it in all asserts from `path === '/'` to `path === Configuration.baseURL` or just set a default baseURL for all tests (which I did)

Comment (marcoow) [inline tests/unit/session-stores/adaptive-test.js]: I see. I'd just set the `cookiePath` property on the created store instance instead ([here](https://github.com/simplabs/ember-simple-auth/blob/master/tests/unit/session-stores/shared/cookie-store-behavior.js#L98)) in these cases so the test is independent of the `Configuration`.

Comment (IgorKvasn) [inline tests/unit/session-stores/adaptive-test.js]: yes, that's probably a better option then to globally force this setting to all tests

**Comment by IgorKvasn** [comment]
hmm there are some regression tests failing (at least that's how I understood Travis) - I guess to make it backwards compatible, cookie path should be  `null` by default (or to be precise - do not add any cookie path at all)

**Comment by IgorKvasn** [comment]
nope, fastboot tests are failing

**Comment by marcoow** [comment]
> I guess to make it backwards compatible, cookie path should be null by default (or to be precise - do not add any cookie path at all)

You are right - defaulting to `/` would actually be a breaking change:
https://tools.ietf.org/html/rfc6265#section-4.1.2.4

**Comment by marcoow** [comment]
@IgorKvasn: can you rebase on `master`?

File: addon/session-stores/cookie.js [hunk 4]
```diff
@@ -218,7 +229,7 @@ export default BaseStore.extend({
     let cookieOptions = {
       domain: this.get('cookieDomain'),
       expires: isEmpty(expiration) ? null : new Date(expiration),
-      path: '/',
+      path: this.get('cookiePath'),
```
Comment (marcoow) [inline addon/session-stores/cookie.js]: Hm, this is actually a breaking change. We'll need to default the property to `'/'` instead.

Comment (marcoow) [inline addon/session-stores/cookie.js]: I fixed it
